tests/unit/DebugUtilsTest.ts

```diff
@@ -381,92 +457,244 @@ describe('Debug Utils', () => {
             expect(() => {
-                DebugUtils.validateObject(DebugUtils.onyxDataToString([{a: 1, b: 'a', c: [], d: {}, e: true, f: false}]), {
-                    a: 'number',
-                    b: 'string',
-                    c: 'array',
-                    d: 'object',
-                    e: 'boolean',
-                    f: 'boolean',
-                });
+                DebugUtils.validateArray(
+                    DebugUtils.onyxDataToString([
+                        {
+                            c: 2,
+                        },
+                    ]),
+                    {
+                        c: 'array',
+                    },
+                );
             }).toThrow();
         });
-        it('throws SyntaxError when value is not a valid JSON - invalid property', () => {
-            expect(() => {
-                DebugUtils.validateObject(DebugUtils.onyxDataToString({a: 'a', b: 'a', c: [], d: {}, e: true, f: false}), {
-                    a: 'number',
-                    b: 'string',
-                    c: 'array',
-                    d: 'object',
-                    e: 'boolean',
-                    f: 'boolean',
+    });
+    describe('validateObject', () => {
+        describe('value is undefined', () => {
+            it('does not throw SyntaxError', () => {
+                expect(() => {
+                    DebugUtils.validateObject('undefined', {});
+                }).not.toThrow();
+            });
+        });
+        describe('value is a JSON representation of an object', () => {
+            describe('object is valid', () => {
+                it('does not throw SyntaxError', () => {
+                    expect(() => {
+                        DebugUtils.validateObject(
+                            DebugUtils.onyxDataToString({
+                                a: 1,
+                                b: 'a',
+                                c: [],
+                                d: {},
+                                e: true,
+                                f: false,
+                            }),
+                            {
+                                a: 'number',
+                                b: 'string',
+                                c: 'array',
+                                d: 'object',
+                                e: 'boolean',
+                                f: 'boolean',
+                            },
+                        );
+                    }).not.toThrow();
                 });
-            }).toThrow();
+            });
+            describe('object has an invalid property', () => {
+                it('throws SyntaxError', () => {
+                    expect(() => {
+                        DebugUtils.validateObject(
+                            DebugUtils.onyxDataToString({
+                                a: 'a',
```

Comment:
`a: 1`, no?

Reply:
@fabioh8010 nope, that property is supposed to be invalid in that test